File: search_ll.c
#include <stdio.h>
#include <stdlib.h>
struct Node { int data; struct Node* next; };
int search(struct Node* head, int key) {
    while (head) {
        if (head->data == key) return 1;
        head = head->next;
    }
    return 0;
}
struct Node* createNode(int x) {
    struct Node* newNode
        = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = x;
    newNode->next = NULL;
    return newNode;
}
int main() {
    struct Node* head = createNode(10);
    head->next = createNode(20);
    head->next->next = createNode(30);
    printf("%s\n", search(head, 20) ? "Found" : "Not Found");
    return 0;
}

File: search_ll.py
class Node:
    def __init__(self, data): self.data = data; self.next = None
def search(head, key):
    while head:
        if head.data == key:
            return True
        head = head.next
    return False
head = Node(10)
head.next = Node(20)
head.next.next=Node(30)
print("Found" if search(head, 20) else "Not Found")
